exasol_integration_test_docker_environment/testing/api_test_environment_context_provider.py

```diff
@@ -66,13 +66,22 @@ def create_context(
         yield spawned
         utils.close_environments(spawned)
 
-    return create_context  # type: ignore
+    return create_context
+
+
+class ApiContextProviderWithTestContainer(Protocol):
+    def __call__(
+        self,
+        name: Optional[str],
+        additional_parameters: Optional[Dict[str, Any]] = None,
+        test_container_content: Optional[TestContainerContentDescription] = None,
+    ) -> ContextManager[ExaslctDockerTestEnvironment]: ...
 
 
 def build_api_context_provider_with_test_container(
     test_environment: ApiTestEnvironment,
     default_test_container_content: TestContainerContentDescription,
-) -> ApiContextProvider:
+) -> ApiContextProviderWithTestContainer:
```

Comment:
I think  `ApiContextProvider` is now redundant, as argument 3 is optional. 
We could integrate these 2 types and potentially also the context providers.

Besides that, the default values for arg `test_container_content` in the following functions are really confusing: 
* `ApiContextProviderWithTestContainer` 
* `build_api_context_provider_with_test_container`
* The context provider returned by `build_api_context_provider_with_test_container`

Reply:
See separate ticket
* https://github.com/exasol/integration-test-docker-environment/issues/482